### PR TITLE
Fix HTTP RequestState values (fixes Pretendo Network support with HLE http)

### PR DESCRIPTION
**Pull request by PabloMK7**
Fixed the `RequestState` enum having the wrong valeus/meanings/documentation. Updated the HTTP HLE methods to use the values properly.

This change fixes Pretendo Network not working properly while using HLE HTTP. Using LLE HTTP is still broken due to a bug in Pretendo Network HTTP patch.

### CODE REVIEW

**Review by DaniElectra — APPROVED** [review]
Looks good

File: src/core/hle/service/http/http_c.h [hunk 1]
```diff
@@ -48,12 +48,16 @@ enum class RequestMethod : u8 {
...
+    NotStarted = 0x1,         // Request has not started yet.
+    ConnectingToServer = 0x5, // Request in progress, connecting to server.
+    SendingRequest = 0x6,     // Request in progress, sending HTTP request.
+    ReceivingResponse = 0x7,  // Request in progress, receiving HTTP response and headers.
+    ReceivingBody = 0x8,      // Request in progress, receiving HTTP body. The HTTP module may
+                              // get stuck in this state if the internal receive buffer gets full.
+                              // Once the user calls ReceiveData it will get unstuck.
+    Received = 0x9,  // Request is finished and all data has been received. HTTP transitions
+                     // to the Completed state shortly afterwards after some cleanup.
+    Completed = 0xA, // Request is completed.
```
Comment (DaniElectra) [inline src/core/hle/service/http/http_c.h:60]: Can you fix up the spacing on here?

Comment (PabloMK7) [reply src/core/hle/service/http/http_c.h]: Done!

**Comment by DaniMKX** [comment]
Nice one!

**Pull request merged**